### PR TITLE
WIP add vvps-engine

### PR DESCRIPTION
**Pull request by normandyxo**
Need to figure out a way for restify to serve static content.

Let me know if this is not the correct way of doing things! I just followed the white rabbit.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,7 @@
 # banana-stand-tunes
-Collaborative Spotify playlist webapp.
+Collaborative playlist webapp.
 
 ### Project Set-Up
-1. Run `npm install` in the root directory.
+1. Run `npm install` in the banana-stand directory.
```
Comment (normandyxo) [inline README.md]: Should be root.

File: banana-stand/app/controllers/song.controller.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+
+/**
+ * Get a shark by its id
+ * @param req
+ * @param res
+ * @param next
+ */
+songController.getOne = function (req, res, next) {
+    Song.findById(req.params.id, function (err, shark) {
```
Comment (normandyxo) [inline banana-stand/app/controllers/song.controller.js]: much copy pasta.

File: banana-stand/config/environments/default.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+          },
+          {
+            "METHOD": "POST",
+            "VALUE": "post"
+          }
+        ]
+      }
+    }
+  },
+  "DB_URI": "mongodb://vvps-mongooser:black0ps@ds039441.mongolab.com:39441/vvps-dev"
```
Comment (normandyxo) [inline banana-stand/config/environments/default.json]: Need to update db string

File: .idea/banana-stand-tunes.iml [hunk 4]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (normandyxo) [inline .idea/banana-stand-tunes.iml]: All this shiz needs to be ignored.

File: banana-stand/config/environments/default.json [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+          {
+            "METHOD": "POST",
+            "VALUE": "post"
+          }
+        ]
+      }
+    }
+  },
+  "DB_URI": "mongodb://loose-seal:banana@ds047901.mongolab.com:47901/banana-stand-tunes"
+}
```
Comment (aeonrex) [inline banana-stand/config/environments/default.json:27]: this

File: banana-stand/server.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    restify =  engine.restify;
+
+// turn da engine on
+engine.bootstrap({
+    name: 'test-api',
+    version: '0.0.1'
+}, function () {
+
+    // default route, return the index.html
+    engine.server.get('/', restify.serveStatic({
```
Comment (aeonrex) [inline banana-stand/server.js:14]: nice

**Comment by aeonrex** [comment]
giphy

**Comment by aeonrex** [comment]
:banana: :ship:

File: banana-stand/public/js/common/common-controllers.module.js [hunk 7]
```diff
@@ -3,4 +3,4 @@
 var ng = require('angular');
 
 module.exports = ng.module('bst.common.controllers', [])
-    .controller('MasterController', require('./controllers/master.controller'));
+    .controller('MasterController', require('./controllers/master.controller.js'));
```
Comment (ksison) [inline banana-stand/public/js/common/common-controllers.module.js:6]: hmm why does this need the extension?

File: vvps-engine/README.md [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+				environments/
+				|
+				anything else.json
+			app/
+				|
+				controllers/
+					<api-controllers>
+			server.js
+
+	In da main just do dis:
```
Comment (ksison) [inline vvps-engine/README.md:25]: luv it

**Comment by ksison** [comment]
SHIP PIT

File: banana-stand/public/js/common/common-controllers.module.js (hunk 7 above)
Comment (normandyxo) [inline banana-stand/public/js/common/common-controllers.module.js:6]: I renamed the file and told Webstorm to update any references to it. I'll update it in #3

**Pull request merged**